esphome/components/sprinkler/__init__.py

```diff
@@ -0,0 +1,449 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome import automation
+from esphome.automation import maybe_simple_id
+from esphome.components import switch
+from esphome.const import (
+    CONF_ID,
+    CONF_NAME,
+    CONF_REPEAT,
+    CONF_RUN_DURATION,
+)
+
+CODEOWNERS = ["@kbx81"]
```

Comment:
If for some reason someone creates a sprinkler config with no actual switches, it will probably pass config, but fail to compile leaving a compile error for the user to figure out

```suggestion
AUTO_LOAD = ["switch"]
CODEOWNERS = ["@kbx81"]
```